test/test_backup.sh

```diff
@@ -0,0 +1,115 @@
+    mysql -u${DB_USER} -p${DB_PASS} -h${DB_HOST} --database ${DB_NAME} -Bse 'TRUNCATE TABLE page'
+}
+
+test_1() {
+    printf "Test that backups are stored in the backup dir\n"
+    # Run backup script, count backup files before and after
+    file_count_before=$(_count_backup_files)
+    /app/backup.sh
+    file_count_after=$(_count_backup_files)
+    
+    # Check that 2 backup files have been created (SQL and XML backups)
+    # Please keep the spacing after [[ and around ==
+    if [[ $(($file_count_before+2)) == $file_count_after ]]; then
+        printf ' - Test backup OK: Two backup files where created.\n'
+    else
+        printf ' - Test backup FAILED: no backup where files created.\n'
+        exit 1
+    fi
+}
+
+test_2() {
+    printf "Test that the database can be restored from SQL backup\n"
+
+    # Check that wiki is running and accessible
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '200' ]]; then
+        printf " - Test restore SQL FAILED: Could not locate wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi
+
+    # Run backup script
+    /app/backup.sh
+
+    # Break the wiki
+    $(_break_wiki)
+    
+    # Check that wiki is broken
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '404' ]]; then
+        printf " - Test restore SQL FAILED: Something went wrong while erasing pages of wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi
+    
+    # Run restore script with defaults
+    /app/restore.sh
+    
+    # Check that wiki is running and accessible
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '200' ]]; then
+        printf " - Test restore SQL FAILED: Could not restore wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi    
+    printf ' - Test restore SQL OK: Wiki was restored from SQL dump.\n'
+}
+
+test_3() {
+    printf "Test that the database can be restored from XML backup\n"
+    
+    # Check that wiki is running and accessible
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '200' ]]; then
+        printf " - Test restore XML FAILED: Could not locate wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi
+    
+    # Run backup script
+    /app/backup.sh
+    
+    # Break the wiki
+    $(_break_wiki)
+    
+    # Check that wiki is broken
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '404' ]]; then
+        printf " - Test restore XML FAILED: Something went wrong while erasing pages of wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi
+    
+    # Run restore script from last XML backup
+    /app/restore.sh -t xml
+    
+    # Check that wiki is running and accessible
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '200' ]]; then
+        printf " - Test restore XML FAILED: Could not restore wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi    
+    printf ' - Test restore XML OK: Wiki was restored from XML dump.\n'
+}
```

Comment:
Suggestion for other test, restore backuped files and also check the accumulated file-size.

Reply:
future work. Maybe we should focus on bugfixes first.